include/sparrow/arrow_interface/arrow_array_schema_utils.hpp

```diff
@@ -21,11 +21,50 @@
 #include <type_traits>
 #include <vector>
 
+#include "sparrow/c_interface.hpp"
 #include "sparrow/utils/memory.hpp"
 #include "sparrow/utils/mp_utils.hpp"
 
 namespace sparrow
 {
+    /**
+     * Release the children and dictionnary of an ArrowArray or ArrowSchema.
+     * 
+     * @tparam T ArrowArray or ArrowSchema
+     * @param t The ArrowArray or ArrowSchema to release.
+     */
+    template <class T>
+    requires std::same_as<T, ArrowArray> || std::same_as<T, ArrowSchema>
+    void release_common_arrow(T* t)
+    {
+        if (t->dictionary)
+        {
+            if (t->dictionary->release)
+            {
+                t->dictionary->release(t->dictionary);
```

Comment:
This can throw. As suggested some time ago in another PR, we probably need a "safe invoke" like in mamba to use here and in `arrow_proxy`'s destructor.

Reply:
The exception is not catched here but in the destructor of the `arrow_proxy`